src/python/pants/backend/experimental/python/register.py

```diff
@@ -1,10 +1,14 @@
 # Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-from pants.backend.experimental.python import user_lockfiles
+from pants.backend.experimental.python import publish_pypi, user_lockfiles
```

Comment:
I think that it would be better to separate the rules for publishing to pypi, from the tool used to do it (twine). That way, there can potentially be multiple tools to choose from, that all publishes to a pypi registry, and it will be up to the user to decide which one to enable.

Reply:
Would the `pypi_repository` target stay as a 'core' type, with the `twine` subsystem being a user-configurable backend, or does `pypi_repository` become `twine_repository` and sit in it's own backend?

My counter argument: 
> There should be one-- and preferably only one --obvious way to do it.
Supporting two tools to publish to pypi just sounds like twice as much code to maintain to achieve the same goal.

Reply:
Yes, not saying that there'd be more than one supported by core pants.

Reply:
> I think that it would be better to separate the rules for publishing to pypi, from the tool used to do it (twine). That way, there can potentially be multiple tools to choose from, that all publishes to a pypi registry, and it will be up to the user to decide which one to enable.

I'm less sure about this... since the destination (PyPI) has a particular API (, auth requirements, etc), various tools to publish to it are likely to be fairly similar in terms of their arguments. Seems like maybe premature generalization.

Reply:
Just consider if I'd rather use, say, poetry to publish my dist instead of twine, how would that work? (for reasons, like using fewer tools/configuration etc)
I may have to provide my own plugin to support using poetry for this, but I'd want to be able to do so, without the default twine implementation tripping me up.

Reply:
You'd have to define a `poetry_pypi_repository` target.